src/gtk/toga_gtk/widgets/switch.py

```diff
@@ -0,0 +1,56 @@
+from gi.repository import Gtk
+from .base import Widget
+
+
+class Switch(Widget):
+    def create(self):
+        self._on_toggle_handler = None
+
+        self.native = Gtk.Box()
+        self.native.interface = self.interface
+
+        self.label = Gtk.Label(xalign=0)
+        self.label.set_line_wrap(True)
+
+        self.switch = Gtk.Switch()
+        self.switch.connect("notify::active", self._on_toggle)
+
+        self.native.pack_start(self.label, True, True, 0)
+        self.native.pack_start(self.switch, False, False, 0)
+        self.native.connect('show', lambda event: self.rehint())
+
+    def _on_toggle(self, widget, state):
+        if self._on_toggle_handler:
+            self._on_toggle_handler(widget)
```

Comment:
Is there a reason for double handling/double storing the handler here? Is there any reason you couldn't use `self.interface.on_toggle`, instead of storing a second copy here?

Reply:
Actually, there is no need at all! In fact, using the interface's handler makes `set_on_select` useless, at least for GTK-based widgets. Thanks for the input!